docs/rfcs/internal-telemetry.md

```diff
@@ -0,0 +1,66 @@
+# Defining guidelines for internal telemetry
+
+## Overview
+
+The Collector supports generating internal telemetry
+that assists end users when operating the Collector. So
+far, much of the telemetry has been added to components as
+needed, without guidelines for component authors to provide
+a consistent experience to end users. The goal of this document
+is to:
+
+- describe the naming and attributes currently in use
+- define consistent units + naming for telemetry emitted by the Collector
+- define the process that should be used to configure new metrics
+
```

Comment:
Could it make sense to articulate a preference that, where possible, metrics are automatically captured via core collector packages, rather than relying fully on component authors?

Reply:
100%, i would much rather that all common telemetry be generated for the component types via helpers (or even better at a higher level like consumer)

Reply:
Added a note here: https://github.com/open-telemetry/opentelemetry-collector/pull/10695/files#diff-9d00dc5bdea981219f46e2b98ed6b89fbccb252cc2fd53e6f7ce820baaa3e9f9R79